features/mbedtls/targets/TARGET_STM/sha256_alt.h

```diff
@@ -0,0 +1,109 @@
+/**
+ * \file sha256_alt.h
+ *
+ * \brief SHA256 hw acceleration (hash function)
+ *
+ * Copyright (c) 2017, STMicroelectronics
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+#ifndef MBEDTLS_SHA256_ALT_H
+#define MBEDTLS_SHA256_ALT_H
+
+#if defined (MBEDTLS_SHA256_ALT)
+
+#include "cmsis.h"
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define MBEDTLS_SHA256_BLOCK_SIZE ((size_t)(64)) // must be a multiple of 4
+/**
+ * \brief          SHA-256 context structure
+ * \note     HAL_HASH_SHA256_Accumulate cannot handle less than 4 bytes, unless it is the last call to the  function
+ *           A MBEDTLS_SHA256_BLOCK_SIZE bytes buffer is used to save values and handle the processing
+ *           MBEDTLS_SHA256_BLOCK_SIZE bytes per MBEDTLS_SHA256_BLOCK_SIZE bytes
+ *           If sha256_finish is called and sbuf_len>0, the remaining bytes are accumulated prior to the call to HAL_HASH_SHA256_Finish
+ */
+typedef struct
+{
+    int is224;                  /*!< 0 => SHA-256, else SHA-224 */  
```

Comment:
@adustm I know. We're stuck with this in the portable code because we don't want to change types that are visible in the public API (it would break binary compatibility). Here the backward compatibility constraint does not apply. But it's extremely minor anyway, it would save 4 bytes per context at most.